SuperRustBoy-iOS/SceneDelegate.swift

```diff
@@ -25,15 +27,30 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         // Create the SwiftUI view that provides the window contents.
         let rootView = RustBoyView(rustBoy: rustBoy)
 
+        picker.delegate = self
+
         // Use a UIHostingController as window root view controller.
         if let windowScene = scene as? UIWindowScene {
             let window = UIWindow(windowScene: windowScene)
             window.rootViewController = UIHostingController(rootView: rootView)
             self.window = window
             window.makeKeyAndVisible()
+
+            window.rootViewController?.present(picker, animated: true)
         }
     }
 
+    func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentsAt urls: [URL]) {
+
+        let nsURL = urls.first! as NSURL
+
+        let romPath = nsURL.resourceSpecifier
+        let saveFilePath = NSTemporaryDirectory() + "save.sav"
```

Comment:
Maybe you should extract the rom file name and add ".sav" onto the end instead of using a single save file for everything.

Reply:
Doing that now and saving it in the documents directory for now.